maskrcnn_benchmark/utils/metric_logger.py

```diff
@@ -61,3 +65,42 @@ def __str__(self):
                 "{}: {:.4f} ({:.4f})".format(name, meter.median, meter.global_avg)
             )
         return self.delimiter.join(loss_str)
+
+
+class TensorboardLogger(MetricLogger):
+    def __init__(self,
+                 log_dir='logs',
+                 exp_name='maskrcnn-benchmark',
+                 start_iter=0,
+                 delimiter='\t'):
+
+        super(TensorboardLogger, self).__init__(delimiter)
+        self.iteration = start_iter
+        self.writer = self._get_tensorboard_writer(log_dir, exp_name)
+
+    @staticmethod
+    def _get_tensorboard_writer(log_dir, exp_name):
+        try:
+            from tensorboardX import SummaryWriter
+        except ImportError:
+            raise ImportError(
+                'To use tensorboard please install tensorboardX '
+                '[ pip install tensorflow tensorboardX ].'
+            )
+
+        if is_main_process():
+            timestamp = datetime.fromtimestamp(time.time()).strftime('%Y%m%d-%H:%M')
+            tb_logger = SummaryWriter('{}/{}-{}'.format(log_dir, exp_name, timestamp))
+            return tb_logger
+        else:
+            return None
+
+    def update(self, ** kwargs):
+        super(TensorboardLogger, self).update(**kwargs)
+        if self.writer:
+            for k, v in kwargs.items():
+                if isinstance(v, torch.Tensor):
+                    v = v.item()
+                assert isinstance(v, (float, int))
+                self.writer.add_scalar(k, v, self.iteration)
+            self.iteration += 1
```

Comment:
I think this iteration update needs to be elsewhere, explicitly, since `do_train` calls `update` twice in the same iteration ([loss update]https://github.com/facebookresearch/maskrcnn-benchmark/pull/163/files#diff-29486803add8a1cde2a6e5b741434c7cR83), [time update](https://github.com/facebookresearch/maskrcnn-benchmark/pull/163/files#diff-29486803add8a1cde2a6e5b741434c7cR91)). Or the semantics of Meters can be changed so that update should only be called once per iteration?

Reply:
yes, this needs to be addressed

Reply:
it should be fine, we are looping on kwargs, so only on the keys that should be updated.